File: internal/connector/connectors/pluginc/testdata/conform.py
#!/usr/bin/env python3
"""Conformance fixture for the Python plugin SDK, driven by sdkconform_test.go
through the real pluginc connector. Fixed rows exercising the cell types,
NULLs, and the SDK's automatic predicate/limit application."""

import datetime
import os
import sys

sys.path.insert(
    0,
    os.path.join(os.path.dirname(os.path.abspath(__file__)), "..", "..", "..", "..", "..", "sdk", "python"),
)
import ttplugin  # noqa: E402

T1 = datetime.datetime(2026, 1, 1, 10, 0, 0, tzinfo=datetime.timezone.utc)
T2 = datetime.datetime(2026, 1, 2, 10, 0, 0, tzinfo=datetime.timezone.utc)

ROWS = [
    [1, 1.5, "alpha", True, T1, b"hi"],
    [2, 2.5, "beta", False, T2, None],
    [3, None, "aloe", True, None, b"yo"],
]

ttplugin.serve(ttplugin.Plugin(
    name="conform",
    datasets={
        "vals": ttplugin.Dataset(
            columns=[
                ttplugin.Column("i", "int"),
                ttplugin.Column("f", "float", nullable=True),
                ttplugin.Column("s", "string"),
                ttplugin.Column("b", "bool"),
                ttplugin.Column("t", "time", nullable=True),
                ttplugin.Column("by", "bytes", nullable=True),
            ],
            rows=lambda req: ROWS,
        ),
    },
))

File: internal/connector/connectors/pluginc/testdata/conform.mjs
#!/usr/bin/env node
// Conformance fixture for the Node.js plugin SDK, driven by sdkconform_test.go
// through the real pluginc connector. Same fixed rows as conform.py.
import { serve } from "../../../../../sdk/node/ttplugin.js";

const T1 = new Date(Date.UTC(2026, 0, 1, 10, 0, 0));
const T2 = new Date(Date.UTC(2026, 0, 2, 10, 0, 0));

const ROWS = [
  [1, 1.5, "alpha", true, T1, Buffer.from("hi")],
  [2, 2.5, "beta", false, T2, null],
  [3, null, "aloe", true, null, Buffer.from("yo")],
];

serve({
  name: "conform",
  datasets: {
    vals: {
      columns: [
        { name: "i", type: "int" },
        { name: "f", type: "float", nullable: true },
        { name: "s", type: "string" },
        { name: "b", type: "bool" },
        { name: "t", type: "time", nullable: true },
        { name: "by", type: "bytes", nullable: true },
      ],
      rows: () => ROWS,
    },
  },
});
